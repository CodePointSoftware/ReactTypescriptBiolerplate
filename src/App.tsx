import React, { useState } from 'react';

type Props = {
    title?: string;
    initialValue?: number;
};

const App: React.FunctionComponent<Props> = ({ title = 'Counter', initialValue = 0 }) => {
    const [counter, setCounter] = useState(initialValue);

    const increment = () => setCounter(counter + 1);
    const decrement = () => setCounter(counter - 1);

    return (
        <>
            <h1>
                This is a Basic Typescript Component with {title}: <span className="counter">{counter}</span>
            </h1>
            <button onClick={increment}>Increment</button>
            <button onClick={decrement}>Decrement</button>
        </>
    );
};

export default App;
