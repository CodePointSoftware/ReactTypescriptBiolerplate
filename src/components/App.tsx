import React from 'react';

class App extends React.Component<{}, {}> {
    render() {
        return <h1>This is a Basic Typescript Component</h1>;
    }
}

export default App;
