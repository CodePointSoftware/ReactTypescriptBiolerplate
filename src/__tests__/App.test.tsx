import * as React from 'react';
import { mount } from 'enzyme';

import App from '../App';

describe('APP', () => {
    let component;
    beforeEach(() => {
        component = mount(<App />);
    });

    it('renders the heading', () => {
        expect(component).toMatchSnapshot();
    });

    it('increments counter properly', () => {
        expect(component.find('.counter').text()).toEqual('0')
        component.find('button').at(0).simulate('click');
        expect(component.find('.counter').text()).toEqual('1')
    });

    it('decrements counter properly', () => {
        expect(component.find('.counter').text()).toEqual('0')
        component.find('button').at(1).simulate('click');
        expect(component.find('.counter').text()).toEqual('-1')
    })
});
