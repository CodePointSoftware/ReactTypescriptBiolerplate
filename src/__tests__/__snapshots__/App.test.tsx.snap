// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP renders the heading 1`] = `
<App>
  <h1>
    This is a Basic Typescript Component with 
    Counter
    : 
    <span
      className="counter"
    >
      0
    </span>
  </h1>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
</App>
`;
